src/dbms/dbms_handler.cpp

```diff
@@ -9,58 +9,214 @@
+  // TODO: Unify durability and wal
   const auto durability_dir = db_dir / ".durability";
+  const auto wal_dir = db_dir / ".wal";
   utils::EnsureDirOrDie(db_dir);
   utils::EnsureDirOrDie(durability_dir);
+  utils::EnsureDirOrDie(wal_dir);
   durability_ = std::make_unique<kvstore::KVStore>(durability_dir);
 
-  // Generate the default database
-  MG_ASSERT(!NewDefault_().HasError(), "Failed while creating the default DB.");
+  /*
+   * DURABILITY
+   */
+  // Migrate durability
+  Durability::Migrate(durability_.get(), root);
+  auto directories = std::set{std::string{kDefaultDB}};
 
   // Recover previous databases
   if (recovery_on_startup) {
-    for (const auto &[name, _] : *durability_) {
-      if (name == kDefaultDB) continue;  // Already set
-      spdlog::info("Restoring database {}.", name);
-      MG_ASSERT(!New_(name).HasError(), "Failed while creating database {}.", name);
+    auto it = durability_->begin(std::string(kDBPrefix));
+    auto end = durability_->end(std::string(kDBPrefix));
+    for (; it != end; ++it) {
+      const auto &[key, config_json] = *it;
+      const auto name = key.substr(kDBPrefix.size());
+      auto json = nlohmann::json::parse(config_json);
+      const auto uuid = json.at("uuid").get<utils::UUID>();
+      const auto rel_dir = json.at("rel_dir").get<std::filesystem::path>();
+      spdlog::info("Restoring database {} at {}.", name, rel_dir);
+      auto new_db = New_(name, uuid, rel_dir);
+      MG_ASSERT(!new_db.HasError(), "Failed while creating database {}.", name);
+      directories.emplace(rel_dir.filename());
       spdlog::info("Database {} restored.", name);
     }
+    // Read the last timestamp
+    auto lcst = durability_->Get(kLSCTSKey);
+    if (lcst) {
+      last_commited_system_timestamp_ = std::stoul(*lcst);
+      system_timestamp_ = last_commited_system_timestamp_;
+    }
   } else {  // Clear databases from the durability list and auth
     auto locked_auth = auth->Lock();
-    for (const auto &[name, _] : *durability_) {
+    auto it = durability_->begin(std::string{kDBPrefix});
+    auto end = durability_->end(std::string{kDBPrefix});
+    for (; it != end; ++it) {
+      const auto &[key, _] = *it;
+      const auto name = key.substr(kDBPrefix.size());
       if (name == kDefaultDB) continue;
       locked_auth->DeleteDatabase(name);
-      durability_->Delete(name);
+      durability_->Delete(key);
+    }
+    // Delete the last timestamp
+    durability_->Delete(kLSCTSKey);
+  }
+
+  /*
+   * DATABASES CLEAN UP
+   */
+  // Clean the unused directories
+  for (const auto &entry : std::filesystem::directory_iterator(db_dir)) {
+    const auto &name = entry.path().filename().string();
+    if (entry.is_directory() && !name.empty() && name.front() != '.') {
+      auto itr = directories.find(name);
+      if (itr == directories.end()) {
+        std::filesystem::remove_all(entry);
+      } else {
+        directories.erase(itr);
+      }
     }
   }
```

Comment:
Is this safe? Some admin forgets to start with the right flag `recovery_on_startup == false` and production databases could be removed from filesystem.

Reply:
This is in line with how we handle other similar situations.